soci/index_builder.go

```diff
@@ -83,17 +86,23 @@ func WithBuildToolIdentifierOption(buildToolIdentifier string) IndexBuilderOptio
 	}
 }
 
-// BuildIndex builds SOCI index for an image
+func WithPlatformOption(platform ocispec.Platform) IndexBuilderOption {
```

Comment:
nit: the community has generally fallen into either calling these functions `WithPlatform` or just `Platform`. If this is going to be our public API can we use a similar convention?

This can be a separate PR, if you agree.

Reply:
Yes completely agree. Currently we have a name conflict for other `WithXXX` functions, so that's why I added the `Option` suffix. @qingshenhuang will rename them in his PR https://github.com/awslabs/soci-snapshotter/pull/169. He's currently waiting for my PR to be merged.